src/plugins/event-plugins/__tests__/NavigationPlugin.test.ts

```diff
@@ -92,13 +93,43 @@ describe('NavigationPlugin tests', () => {
         // enables plugin by default
         const plugin: NavigationPlugin = buildNavigationPlugin();
 
-        // adds eventListener on load event
-        plugin.load(context);
+        // window has already loaded, so it will fire when we load.
+        // thus disabling the plugin before we load
         plugin.disable();
+        plugin.load(context);
+
         window.dispatchEvent(new Event('load'));
         plugin.disable();
 
         // Assert
         expect(record).toHaveBeenCalledTimes(0);
     });
+
+    test('when page has not loaded then we do not fire until load is completed', async () => {
+        // Setting up new mocked window that has not loaded
+        (window as any).performance = performanceEventNotLoaded.performance();
+        (window as any).PerformanceObserver =
+            performanceEvent.PerformanceObserver;
```

Comment:
Do we need to mock `PerformanceObserver` here? It was removed from `NavigationPlugin`

Reply:
Agreed, no need. Will reflect on next revision.